### PR TITLE
add stale gha

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR adds the `stale` GHA in order to help maintain a healthy issues and prs backlog.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by bjlittle** [comment]
I guess the `days-before-stale` and `days-before-close` are up for debate.

If it helps, here is a plot of number of issues that would be not be considered state for the given dates or days since now `3 Aug 2021` e.g., for the proposed `days-before-stale` of `1000` which is `7 Nov 2018`, approximately `159` issues would remain **not** marked as `stale` out of `295`.

stale_issues

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+            Otherwise this PR will be automatically closed in 28 days time.
+
+          # Comment on the staled issues while closed.
+          close-issue-message: |
+            This stale issue has been automatically closed due to a lack of community activity.
+            
+            If you still care about this issue, then please either:
+              * Re-open this issue, if you have sufficient permissions, or
+              * Add a comment pinging `@SciTools/iris-devs` who will re-open on your behalf.
```
Comment (bjlittle) [inline .github/workflows/stale.yml:42]: To be honest, we could have a GHA `bot` that auto-opens closed issues/prs whenever the user adds a comment like `@SciTools/iris-devs, please re-open` akin to `@conda-forge-admin, please rerender` on a `conda-forge` feedstock.

**Comment by jamesp** [comment]
I think this is a great idea.  I'd be all in favour of a shorter `days-before-stale`. Realistically, if we haven't addressed it in 365 days it's probably never going to get addressed, and the project will have moved on significantly from this point of raising.

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+          # Label to apply on staled issues. 
+          stale-issue-label: Stale
+
+          # Label to apply on staled prs.
+          stale-pr-label: Stale
+
+          # Labels on issues exempted from stale.
+          exempt-issue-labels: |
+            "Status: Blocked,Status: Decision Required,Status: Needs Info,Status: Stalled,Peloton 🚴‍️"
```
Comment (jamesp) [inline .github/workflows/stale.yml:60]: I would argue that "Needs Info" and "Stalled" should not be in this list.  They should be staled out if they go 1000 days without info or unstalling

Comment (bjlittle) [reply .github/workflows/stale.yml:60]: Cool, I'd go with that... did I miss out any labels that you'd consider should be in the list?

Comment (bjlittle) [reply .github/workflows/stale.yml]: Done :+1:

**Review by jamesp — APPROVED** [review]

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+name: Stale issues and pull-requests
+on:
+  schedule:
+    - cron: 0 * * * *
```
Comment (jamesp) [inline .github/workflows/stale.yml:4]: Does this need to run every hour?  Could we run every day?

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          days-before-stale: 500
+
+          # Idle number of days before closing stale issues/prs.
+          days-before-close: 28
+
+          # Comment on the staled issues.
+          stale-issue-message: |
+            In order to maintain a backlog of relevant issues, we automatically label them as stale after 500 days of inactivity.
+
+            If this issue is still important to you, then please comment on this issue or remove the stale label.
```
Comment (rcomer) [inline .github/workflows/stale.yml:24]: I think only core devs can remove the label.  How about "If this issue is still important to you, then please comment on this issue and the stale label will be removed." ?

File: .github/workflows/stale.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+            Otherwise this issue will be automatically closed in 28 days time.
+
+          # Comment on the staled prs.
+          stale-pr-message: | 
+            In order to maintain a backlog of relevant PRs, we automatically label them as stale after 500 days of inactivity.
+
+            If this PR is still important to you, then please comment on this PR or remove the stale label.
```
Comment (rcomer) [inline .github/workflows/stale.yml:32]: As above.

File: .github/workflows/stale.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+name: Stale issues and pull-requests
+on:
+  schedule:
+    - cron: 0 0 * * *
```
Comment (bjlittle) [inline .github/workflows/stale.yml:4]: Run GHA daily, see https://crontab.guru/every-day

**Review by rcomer — COMMENTED** [review]
Over on the discussion, there was talk of starting with a 3 year lead, and then gradually reducing.  I think starting with a smaller number of stale issues makes sense, because we can see what kind of activity (if any) is generated by people re-engaging.  It would be a shame if someone did re-engage but we didn't have time to address their comments.

On the gap between stale and closed, I would have said at least a month, as it gives time for people who are on holiday or otherwise away.  So 28 days seems OK to me, but I wouldn't want to go any lower.

**Comment by bjlittle** [comment]
@jamesp Did you want a `whatsnew` entry? I'm really not fussed...

**Comment by jamesp** [comment]
No I don't think it's necessary, thanks @bjlittle!

**Pull request merged**